packages/json-rpc/sm-provider/src/sm-provider.ts

```diff
@@ -22,8 +28,17 @@ export const getSmProvider = (chain: Chain | Promise<Chain>): JsonRpcProvider =>
           let message = ""
           try {
             message = await resolvedChain.nextJsonRpcResponse()
-          } catch (e) {
-            if (listening) onError()
+          } catch (error) {
+            if (listening)
+              onError({
```

Comment:
@josepot, following up on our previous discussion, you suggested to simply not call `onError` if the error is not recoverable, since if and when this error happens (which in theory it should never), there's realistically nothing that we could do.

However, in practice Smoldot does blow up more often than not, and in that case, I believe it's better to blow up with an error & let the application decide how to handle it, rather than do nothing & leaving it in a frozen/inactive state.

It’s really a case of "pick your poison" 😅. Curious to hear your thoughts on this!